backend/pkg/config/config.go

```diff
@@ -0,0 +1,202 @@
+package config
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/knadh/koanf"
+	"github.com/knadh/koanf/providers/basicflag"
+	"github.com/knadh/koanf/providers/env"
+)
+
+// defautl port is 4466.
+const defaultPort = 4466
+
+type Config struct {
+	InCluster        bool   `koanf:"in-cluster"`
+	DevMode          bool   `koanf:"dev"`
+	InsecureSsl      bool   `koanf:"insecure-ssl"`
+	KubeConfigPath   string `koanf:"kubeconfig"`
+	StaticDir        string `koanf:"html-static-dir"`
+	PluginsDir       string `koanf:"plugins-dir"`
+	BaseURL          string `koanf:"base-url"`
+	Port             uint   `koanf:"port"`
+	ProxyURLs        string `koanf:"proxy-urls"`
+	OidcClientID     string `koanf:"oidc-client-id"`
+	OidcClientSecret string `koanf:"oidc-client-secret"`
+	OidcIdpIssuerURL string `koanf:"oidc-idp-issuer-url"`
+	OidcScopes       string `koanf:"oidc-scopes"`
+}
+
+func (c *Config) Validate() error {
+	if !c.InCluster && (c.OidcClientID != "" || c.OidcClientSecret != "" || c.OidcIdpIssuerURL != "") {
+		return errors.New(`oidc-client-id, oidc-client-secret, oidc-idp-issuer-url flags 
+		are only meant to be used in inCluster mode`)
+	}
+
+	if c.BaseURL != "" && !strings.HasPrefix(c.BaseURL, "/") {
+		return errors.New("base-url needs to start with a '/' or be empty")
+	}
+
+	return nil
+}
+
+// Parse Loads the config from flags and env.
+// env vars should start with HEADLAMP_CONFIG_ and use _ as separator
+// If a value is set both in flags and env then env takes priority.
```

Comment:
Hmm, is this the default behavior in most programs?
I was under the impression that it's be default-config-file < env vars < flags .

Reply:
I have quickly checked and it seems that Azure and AWS CLIs do assign a lower priority to env vars than flags. The reasoning may be that usually when you run a command directly, you're hardcoding a value/option you want to use, not the default or the environment predefined ones.

Reply:
Changed the priority order